src/xrpld/app/misc/CredentialHelpers.cpp

```diff
@@ -225,6 +226,50 @@ makeSorted(STArray const& in)
     return out;
 }
 
+NotTEC
+checkArray(STArray const& credentials, unsigned maxSize, beast::Journal j)
+{
+    if (credentials.empty() || (credentials.size() > maxSize))
+    {
+        JLOG(j.trace()) << "Malformed transaction: "
+                           "Invalid credentials size: "
+                        << credentials.size();
+        return temMALFORMED;
+    }
+
+    std::unordered_set<uint256> duplicates;
+    for (auto const& credential : credentials)
+    {
+        auto const& issuer(credential[sfIssuer]);
```

Comment:
Shouldn't you use the `~` operator here? Is it guaranteed to have an `sfIssuer` field all the time?

Reply:
It's guaranteed, because of `InnerObjectFormats`.